Automattic/jetpack
# Add blogroll item block

Add blogroll item block.
This will later be used in the blogroll to show the recommendations.

This PR only provides the scaffolding.  Block updates will be addressed in later PRs.

Fixes #
https://github.com/Automattic/wp-calypso/issues/80909

## Proposed changes:
* Add blogroll item block

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
- Apply this patch:
```
diff --git a/projects/plugins/jetpack/extensions/blocks/blogroll/edit.js b/projects/plugins/jetpack/extensions/blocks/blogroll/edit.js
index aa4901871c..5c0a55deed 100644
--- a/projects/plugins/jetpack/extensions/blocks/blogroll/edit.js
+++ b/projects/plugins/jetpack/extensions/blocks/blogroll/edit.js
@@ -1,4 +1,4 @@
-import { BlockIcon, InspectorControls, useBlockProps } from '@wordpress/block-editor';
+import { BlockIcon, InspectorControls, useBlockProps, InnerBlocks } from '@wordpress/block-editor';
 import {
 	Button,
 	CheckboxControl,
@@ -56,7 +56,10 @@ export function BlogRollEdit( { className, attributes, setAttributes, isSelected
 	 *
 	 * @returns {object} The UI displayed when user edits this block.
 	 */
-
+		// Default template is single WhatsApp block until we offer
+		// more services
+	const DEFAULT_TEMPLATE = [ [ 'jetpack/blogroll-item', {} ] ];
+	const ALLOWED_BLOCKS = [ 'jetpack/blogroll-item' ];
 	return (
 		<div { ...useBlockProps() } className={ className }>
 			{ showPlaceholder && (
@@ -92,6 +95,8 @@ export function BlogRollEdit( { className, attributes, setAttributes, isSelected
 				/>
 			) }
 
+			<InnerBlocks template={ DEFAULT_TEMPLATE } allowedBlocks={ ALLOWED_BLOCKS } />
+
 			<InspectorControls>
 				<PanelBody title={ __( 'Settings', 'jetpack' ) }>
 					<ToggleControl

```
- Add the blogroll block to a post
- Using the quick appender ensure that blogroll items can be added
<img width="695" alt="Screenshot 2023-08-31 at 15 07 57" src="https://github.com/Automattic/jetpack/assets/7000684/16398f4b-11d1-45e0-ba7c-9e025ebfee8d">
